app/src/main/java/org/simple/clinic/editpatient/PatientEditScreenController.kt

```diff
@@ -178,58 +158,62 @@ class PatientEditScreenController @Inject constructor(
         prefillPatientDateOfBirth)
   }
 
-  private fun ongoingEntryPatientEntryChanges(events: Observable<UiEvent>): Observable<UiEvent> {
-    val nameChanges = events
-        .ofType<PatientEditPatientNameTextChanged>()
-        .map { it.name }
-
-    val genderChanges = events
-        .ofType<PatientEditGenderChanged>()
-        .map { it.gender }
-
-    val colonyOrVillageChanges = events
-        .ofType<PatientEditColonyOrVillageChanged>()
-        .map { it.colonyOrVillage }
-
-    val districtChanges = events
-        .ofType<PatientEditDistrictTextChanged>()
-        .map { it.district }
-
-    val stateChanges = events
-        .ofType<PatientEditStateTextChanged>()
-        .map { it.state }
-
-    val phoneNumberChanges = events
-        .ofType<PatientEditPhoneNumberTextChanged>()
-        .map { it.phoneNumber }
-
-    val ageChanges = events
-        .ofType<PatientEditAgeTextChanged>()
-        .map { EntryWithAge(it.age) as OngoingEditPatientEntry.EitherAgeOrDateOfBirth }
+  private fun mergeWithOngoingEntryPatientEntryChanges(): ObservableTransformer<UiEvent, UiEvent> {
```

Comment:
I'm happy that this will force us to use Transformers for every transformation. I was slowly realizing that this is better.